fetchai/ledger/genesis/genesis.py

```diff
@@ -0,0 +1,122 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+import json
+import hashlib
+import base64
+import base58
+import time
+
+from fetchai.ledger.crypto import Entity
+
+def calc_resource_id(resource_address):
+    hasher = hashlib.sha256()
+    hasher.update(resource_address.encode())
+    return base64.b64encode(hasher.digest()).decode()
+
+def generate_token_address(muddle_address):
+    raw_muddle_address = base64.b64decode(muddle_address)
+
+    hasher1 = hashlib.sha256()
+    hasher1.update(raw_muddle_address)
+    token_address_base = hasher1.digest()
```

Comment:
This pattern is so common that it's worth encapsulating, first in this file, and with a future refactor perhaps in a crypto_utilities module?

Reply:
Yes, you are right